### ISSUE TITLE
Wrong stack order when names a mix of text only and numbers only

### ISSUE THREAD

**Issue by raphaelgera** [c3js/c3#1534]
The below code breaks the order and puts the number only name first 

```js
    data: {
        columns: [
            ['data1', 130, 200, 320, 400, 530, 750],
            ['2', 130, 10, 130, 200, 150, 250],
            ['data3', 130, 50, 10, 200, 250, 150]
        ],
        type: 'bar',
        groups: [
            ['data1', '2', 'data3']
        ] 
   }
```

![image](https://cloud.githubusercontent.com/assets/12085354/12242301/4bbf6ffc-b866-11e5-9de6-9f5f3a9260b5.png)
This code works as expected

```js
    data: {
        columns: [
            ['data1', 130, 200, 320, 400, 530, 750],
            ['2data', 130, 10, 130, 200, 150, 250],
            ['data3', 130, 50, 10, 200, 250, 150]
        ],
        type: 'bar',
        groups: [
            ['data1', '2data', 'data3']
        ] 
   }
```

![image](https://cloud.githubusercontent.com/assets/12085354/12242283/2f533c4a-b866-11e5-9527-dc7a4246b51d.png)

**Comment by Misophistful** [comment]
Probably related to the same issue:

When the column title is a number (or a string that just contains a number), then the given order of the columns is ignored, and they're rendered in ascending numerical order of the column title.

This renders the columns in order they're listed (i.e. 'data5', 'data1', 'data2', 'data4', 'data3')
```Javascript
var chart = c3.generate({
    data: {
        columns: [
            ['data5', 30],
            ['data1', 130],
            ['data2', 35],
            ['data4', 60],
            ['data3', 100]
        ],
        type: 'bar'
    }
});
```

This renders the columns in ascending order of their titles (i.e. 1, 2, 3, 4, 5):
```Javascript
var chart = c3.generate({
    data: {
        columns: [
            [5, 30],
            [1, 130],
            [2, 35],
            [4, 60],
            [3, 100]
        ],
        type: 'bar'
    }
});

**Comment by chrisjensen** [comment]
For those stumbling on this looking for a quick fix, I append a space to my column headers.
eg
```
columns: [
['data1', 130, 200, 320, 400, 530, 750],
['2 ', 130, 10, 130, 200, 150, 250],
['data3', 130, 50, 10, 200, 250, 150]
],

```
Here's an example:
http://jsfiddle.net/gur3q5nt/

**Comment by panthony** [comment]
The cause of this issue it the pivot format used between the input data and the generation of the targets.

It uses the names as object keys.

It "works" when the names are alphanumerical but not at all of the names is numeric.

**Comment by kt3k** [comment]
Thanks for the issue. This has been fixed at v0.5.3 by #2314 .
